core/server/common/src/main/java/alluxio/master/journal/JournalUtils.java

```diff
@@ -214,5 +220,123 @@ public static void sinkFlush(Supplier<Set<JournalSink>> journalSinks) {
+    for (alluxio.proto.journal.Journal.JournalEntry oldEntry : entries) {
+      if (oldEntry.hasInodeFile()) {
+        fileEntryMap.put(oldEntry.getInodeFile().getId(),
+            File.InodeFileEntry.newBuilder(oldEntry.getInodeFile()));
+      } else if (oldEntry.hasUpdateInode()) {
+        File.UpdateInodeEntry entry = oldEntry.getUpdateInode();
+        File.InodeFileEntry.Builder builder = fileEntryMap.get(entry.getId());
+        if (builder == null) {
+          newEntries.add(oldEntry);
+          continue;
+        }
+        if (entry.hasAcl()) {
+          builder.setAcl(entry.getAcl());
+        }
+        if (entry.hasCreationTimeMs()) {
+          builder.setCreationTimeMs(entry.getCreationTimeMs());
+        }
+        if (entry.hasGroup() && !entry.getGroup().isEmpty()) {
+          builder.setGroup(entry.getGroup());
+        }
+        if (entry.hasLastModificationTimeMs() && (entry.getOverwriteModificationTime()
+            || entry.getLastModificationTimeMs() > builder.getLastModificationTimeMs())) {
+          builder.setLastModificationTimeMs(entry.getLastModificationTimeMs());
+        }
+        if (entry.hasLastAccessTimeMs() && (entry.getOverwriteAccessTime()
+            || entry.getLastAccessTimeMs() > builder.getLastAccessTimeMs())) {
+          builder.setLastAccessTimeMs(entry.getLastAccessTimeMs());
+        }
+        if (entry.hasMode()) {
+          builder.setMode((short) entry.getMode());
+        }
+        if (entry.getMediumTypeCount() != 0) {
+          builder.clearMediumType();
+          builder.addAllMediumType(new HashSet<>(entry.getMediumTypeList()));
+        }
+        if (entry.hasName()) {
+          builder.setName(entry.getName());
+        }
+        if (entry.hasOwner() && !entry.getOwner().isEmpty()) {
+          builder.setOwner(entry.getOwner());
+        }
+        if (entry.hasParentId()) {
+          builder.setParentId(entry.getParentId());
+        }
+        if (entry.hasPersistenceState()) {
+          builder.setPersistenceState(entry.getPersistenceState());
+        }
+        if (entry.hasPinned()) {
+          builder.setPinned(entry.getPinned());
+        }
+        if (entry.hasTtl()) {
+          builder.setTtl(entry.getTtl());
+        }
+        if (entry.hasTtlAction()) {
+          builder.setTtlAction(entry.getTtlAction());
+        }
+        if (entry.hasUfsFingerprint()) {
+          builder.setUfsFingerprint(entry.getUfsFingerprint());
+        }
+        if (entry.getXAttrCount() > 0) {
+          builder.clearXAttr();
+          builder.putAllXAttr(new HashMap<>(entry.getXAttrMap()));
+        }
+      } else if (oldEntry.hasUpdateInodeFile()) {
+        File.UpdateInodeFileEntry entry = oldEntry.getUpdateInodeFile();
+        File.InodeFileEntry.Builder builder = fileEntryMap.get(entry.getId());
+        if (builder == null) {
+          newEntries.add(oldEntry);
+          continue;
+        }
+        if (entry.hasPersistJobId()) {
```

Comment:
Similarly, can we reuse the logic from `MutableInodeFile.updateFromEntry`?

Reply:
see above